### PR TITLE
TfLite unpack_test test case refactor

### PR DESCRIPTION
**Pull request by Dayananda-V**
Test case refactor changes.

### CODE REVIEW

File: tensorflow/lite/kernels/unpack_test.cc [hunk 1]
```diff
@@ -67,192 +66,72 @@ class UnpackOpModel : public SingleOpModel {
...
-  model.Invoke();
-
-  // Check outputs shapes.
-  const std::vector<std::vector<int>>& output_shapes = model.GetOutputShapes();
-  EXPECT_EQ(output_shapes.size(), 2);
-  EXPECT_THAT(output_shapes[0], ElementsAre(3));
-  EXPECT_THAT(output_shapes[1], ElementsAre(3));
+// float32 tests.
+TEST(UnpackOpTest, FloatThreeOutputs) {
+  Check<float>({3, 2}, 0, {1, 2, 3, 4, 5, 6}, {{2}, {2}, {2}},
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/unpack_test.cc]: can you add args documentation? thanks

Comment (Dayananda-V) [reply tensorflow/lite/kernels/unpack_test.cc]: @renjie-liu 

added args documentation, please check once again, TIA.

File: tensorflow/lite/kernels/unpack_test.cc [hunk 2]
```diff
@@ -67,192 +66,97 @@ class UnpackOpModel : public SingleOpModel {
   std::vector<int> outputs_;
 };
 
-// float32 tests.
-TEST(UnpackOpTest, FloatThreeOutputs) {
-  UnpackOpModel<float> model({TensorType_FLOAT32, {3, 2}}, 0);
-  model.SetInput({1, 2, 3, 4, 5, 6});
-  model.Invoke();
+template <typename T>
+void Check(int axis, std::initializer_list<int> input_shape,
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/unpack_test.cc]: try const & whenever possible, thanks

Comment (Dayananda-V) [reply tensorflow/lite/kernels/unpack_test.cc]: @renjie-liu 

comments taken care.

File: tensorflow/lite/kernels/unpack_test.cc [hunk 3]
```diff
@@ -67,192 +66,97 @@ class UnpackOpModel : public SingleOpModel {
   std::vector<int> outputs_;
 };
 
-// float32 tests.
-TEST(UnpackOpTest, FloatThreeOutputs) {
-  UnpackOpModel<float> model({TensorType_FLOAT32, {3, 2}}, 0);
-  model.SetInput({1, 2, 3, 4, 5, 6});
-  model.Invoke();
+template <typename T>
+void Check(const int axis, const std::initializer_list<int> input_shape,
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/unpack_test.cc]: does const reference not work?

Comment (Dayananda-V) [reply tensorflow/lite/kernels/unpack_test.cc]: const reference fix the changes, sorry for delay.

File: tensorflow/lite/kernels/unpack_test.cc (hunk 2 above)
Comment (renjie-liu) [reply tensorflow/lite/kernels/unpack_test.cc]: looks like input_shape and expected_output_shape can be const reference as well?

**Review by renjie-liu — CHANGES_REQUESTED** [review]

**Comment by Dayananda-V** [comment]
@renjie-liu 

const reference changes for arg parameter handle, please check once again, TIA.

File: tensorflow/lite/kernels/unpack_test.cc [hunk 4]
```diff
@@ -67,192 +66,97 @@ class UnpackOpModel : public SingleOpModel {
...
-TEST(UnpackOpTest, FloatThreeOutputs) {
-  UnpackOpModel<float> model({TensorType_FLOAT32, {3, 2}}, 0);
-  model.SetInput({1, 2, 3, 4, 5, 6});
-  model.Invoke();
+template <typename T>
+void Check(int axis, const std::initializer_list<int>& input_shape,
+           const std::initializer_list<T>& input_data,
+           const std::vector<std::vector<int>>& exp_output_shape,
+           const std::vector<std::vector<T>>& exp_output_data,
+           const TensorType& type = TensorType_FLOAT32) {
```
Comment (renjie-liu) [inline tensorflow/lite/kernels/unpack_test.cc]: nit: for integer, it's actually fine not to do reference

**Review by renjie-liu — DISMISSED** [review]

File: tensorflow/lite/kernels/unpack_test.cc (hunk 4 above)
Comment (Dayananda-V) [reply tensorflow/lite/kernels/unpack_test.cc]: follow the other existing design changes, need to change the reference?

Comment (renjie-liu) [reply tensorflow/lite/kernels/unpack_test.cc]: up to you, I'm just saying it's not required for integer

Comment (Dayananda-V) [reply tensorflow/lite/kernels/unpack_test.cc]: okay

**Comment by Dayananda-V** [comment]
Corrupt PR is replaced with #27371

**Pull request closed without merging**